File: src/main/kotlin/ru/havlong/cards/store/entity/Sample.java
package ru.havlong.cards.store.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.havlong.cards.api.dto.SampleDTO;

import javax.persistence.*;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "sample")
public class Sample {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    public static void fromDTO(SampleDTO dto, Sample ent) {

        ent.setId(dto.getId());
    }
}

File: src/main/kotlin/ru/havlong/cards/models/Sample.kt
package ru.havlong.cards.models

import ru.havlong.cards.transfer.SampleDTO
import javax.persistence.*

@Entity
@Table(name = "sample")
data class Sample(@Id @GeneratedValue(strategy = GenerationType.SEQUENCE) var id: Long?) {

    constructor() : this(null)

    companion object {
        @JvmStatic
        fun from(sample: SampleDTO): Sample = Sample(sample.id)
    }
}
